geode-core/src/main/java/org/apache/geode/cache/client/PoolFactory.java

```diff
@@ -425,6 +433,21 @@
    */
   public PoolFactory setSubscriptionMessageTrackingTimeout(int messageTrackingTimeout);
 
+  /**
+   * A server has an inactivity monitor that ensures a message is sent to a client at least once a
+   * minute (60,000 milliseconds). If a subscription timeout multipler is set in the client it
+   * enables timing out of the subscription feed with failover to another server.
+   * <p>
+   * A value of zero (the default) disables timeouts
+   * <p>
+   * A value of one will time out the server connection after one of its ping intervals (not
+   * recommended)
+   * <p>
+   * A value of two or more will time out the server connection after that many ping intervals have
+   * elapsed
+   */
+  public PoolFactory setSubscriptionTimeoutMultiplier(int multiplier);
```

Comment:
Given that not everyone reads comments that carefully, and given that using a multiplier of 1 is such a horrible idea, I think we should do more to dissuade the user from using this value.  We should either (a) add a small amount to the timeout to account for delays; (b) log a message (warning?) when this is called with a 1; or (c) just disallow a value of 1.  I have a strong preference for approach (a), since this also makes more sense with higher multipliers (a multiplier of 2 should mean 2 missed pings, but we'd expect to hit the timeout before processing a second ping if we ever missed just one).

Reply:
I've revised this and implemented your (a) option